### PR TITLE
CI: Update breaking changes workflow to compare against main

### LINKED ISSUE #44102: Breaking Changes: Compare against main instead of Canary packages
The breaking changes workflow currently compares a PR against the latest canary build of the Grafana NPM packages. However when a change to one of the NPM packages is merged to master we have to wait for the build to succeed before the canary package is published. During this time where main and the latest canary packages are out of sync PRs can be tagged incorrectly as having breaking changes. 

Rather than compare current PR changes against latest canary we should compare them to the built packages in the main branch.

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
Currently we compare changes in a PR against the latest canary npm package however there is a period of time where PRs build against an outdated canary package (due to merge to main and waiting for that to build and publish) which results in breaking changes being incorrectly labelled in PRs.

- [x] build packages for a PR branch and main
- [x] parallelise package building
- [x] use built package artifacts to compare changes
- [x] cache dependency installation

**Which issue(s) this PR fixes**:

Fixes #44102

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mckn — APPROVED** [review]
Wow, great work on this!! ⭐

File: scripts/check-breaking-changes.sh [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
     # Calculate current and previous package paths / names
-    PREV="$PACKAGE_NAME@canary"
-    CURRENT="$PACKAGE_PATH/dist/"
+    PREV="./main/packages/$PACKAGE_PATH/dist/"
+    CURRENT="./pr/packages/$PACKAGE_PATH/dist/"
 
-    # Temporarily skipping @grafana/toolkit, as it doesn't have any exposed static typing
-    if [[ "$PACKAGE_NAME" == '@grafana/toolkit' ]]; then
+    # Temporarily skipping these packages as they don't have any exposed static typing
+    if [[ "$PACKAGE_PATH" == 'grafana-toolkit' || "$PACKAGE_PATH" == 'jaeger-ui-components' ]]; then
```
Comment (leventebalogh) [inline scripts/check-breaking-changes.sh:19]: **Nit:** maybe we could define these as an array at the top of the file?

Like that maybe it is a bit easier to locate and extend.

**Review by leventebalogh — APPROVED** [review]
Amazing 👏

File: scripts/check-breaking-changes.sh (hunk 1 above)
Comment (leventebalogh) [reply scripts/check-breaking-changes.sh:19]: (Actually ignore this for now, it's something that we can refactor later).

**Pull request merged**